File: src/java/eu/sig/training/ch10/PerfectPicture.java
package eu.sig.training.ch10;

import java.awt.Image;

import eu.sig.training.ch06.simpledigitalcamera.SimpleDigitalCamera;

public class PerfectPicture {
    public static SimpleDigitalCamera camera = null;

    // tag::takePerfectPicture[]
    public final static int DAYLIGHT_START = 6;

    public Image takePerfectPicture(int currentHour) {
        Image image;
        if (currentHour < PerfectPicture.DAYLIGHT_START) {
            camera.flashLightOn();
            image = camera.takeSnapshot();
            camera.flashLightOff();
        } else {
            image = camera.takeSnapshot();
        }
        return image;
    }
    // end::takePerfectPicture[]
}

File: src/csharp/eu/sig/training/ch10/PerfectPicture.cs
using System.Drawing;
using eu.sig.training.ch06.simpledigitalcamera;

namespace eu.sig.training.ch10
{
    public class PerfectPicture
    {
        public static ISimpleDigitalCamera camera = null;

        // tag::takePerfectPicture[]
        public const int DAYLIGHT_START = 6;

        public Image TakePerfectPicture(int currentHour)
        {
            Image image;
            if (currentHour < PerfectPicture.DAYLIGHT_START)
            {
                camera.FlashLightOn();
                image = camera.TakeSnapshot();
                camera.FlashLightOff();
            }
            else
            {
                image = camera.TakeSnapshot();
            }
            return image;
        }
        // end::takePerfectPicture[]
    }
}
